### ISSUE TITLE
Error on compiling

### ISSUE THREAD

**Issue by iamthom** [tobiasBora/MT7630E_3.16#3]
Hello, 

I'm trying to compiling this source in Kernel 3.16, but no luck. Here is the error I got :
![kernel_316](https://cloud.githubusercontent.com/assets/5789364/6143526/e5f43236-b1fc-11e4-87fe-124eaef69db1.png)

My laptop is Ubuntu 14.04, with kernel 3.13.0-32-generic

Thanks.

**Comment by tobiasBora** [comment]
Hello Tom,

I'm not sure it will work, but please could you try to install the headers from the kernel sources ? It should looks like :

sudo apt-get install linux-
headers-$(uname -r)


-- 
Envoyé de mon téléphone Android avec K-9 Mail. Excusez la brièveté.

**Comment by iamthom** [comment]
Hey, sorry for the late response.

It gimme message like this :

`linux-headers-3.13.0-32-generic is already the newest version.`
`linux-headers-3.13.0-32-generic set to manually installed.`
`0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.`

I wonder why I should install a kernel which has already installed.

btw, in order to prevent the system to update the highest kernel I prevented the apt, by set it in /etc/apt/preferences.

I attached an image of it.

![install_kernel](https://cloud.githubusercontent.com/assets/5789364/6161895/5e252f5e-b2ab-11e4-8eeb-f2916cb51124.png)

**Comment by tobiasBora** [comment]
I'm sorry but there is a really strange thing in your commands: sometimes "uname -r" gives 3.16.0-031600-lowlatency (see your first message) and sometimes it returns 3.13.0-32-generic. Are you changing your kernel ?

The idea is that the file "rt2x00/Makefile" ask for the kernel version with "uname -r", and then it needs to go in /lib/modules/<kernel version>/build to find all the files it needs to compile the driver. This folder is created when you install the source packages linux-headers-<kernel version>, as you can see in this page: http://packages.ubuntu.com/trusty-updates/i386/linux-headers-3.16.0-30-generic/filelist.

So choose a kernel (3.13 generic or 3.16 lowlatency), run
   sudo apt-get install linux-headers-$(uname -r)
and then you can compile the MT7630E_3.16 module with
  make
and if it doesn't return any error:
  make load
  make install

If you still have an error please return the result of
   uname -r
   ls /lib/modules/
   ls /lib/modules/$(uname -r)/
   ls /lib/modules/$(uname -r)/build

**Comment by iamthom** [comment]
Yes. I changed my kernel. In the first pic, I was using 3.16, and because I still had no luck, I reverted back tp 3.13. I'll try it later then.
